pkg/providers/pricing/pricing.go

```diff
@@ -80,6 +79,19 @@ type zonal struct {
 	prices       map[string]float64
 }
 
+func combineZonalPricing(pricingData ...zonal) zonal {
+	c := newZonalPricing(0)
+	for _, elem := range pricingData {
+		if elem.defaultPrice != 0 {
+			c.defaultPrice = elem.defaultPrice
+		}
+		for k, v := range elem.prices {
+			c.prices[k] = v
+		}
```

Comment:
nit: can we name the key and value here? It's not clear what these are from the context of the function alone.
```suggestion
		for zone, price := range elem.prices {
			c.prices[zone] = price
		}
```